File: Challenge 1.py
'''
Write a program that will display a joke
Don’t display the punchline until the reader
hits the enter key.
Extension
display the punchline in a different colour
'''

from termcolor import colored
import keyboard

print('If you want me to tell you a joke, press ENTER.. if you don\'t like jokes, press any other key.')
while True:
    try:  
        if keyboard.is_pressed('enter'):
            print(colored('Why don\'t scientists trust atoms?', 'red'))
            print(colored('Because they make up everything.', 'green'))
            break
        elif keyboard.read_key() != 'enter':
            break
    except:
        break

File: Challenge 1.java
import java.util.Scanner;

class Joke {
    public static final String ANSI_RESET = "\u001B[0m";
    public static final String ANSI_RED = "\u001B[31m";
    public static final String ANSI_GREEN = "\u001B[32m";
    public static void main(String[] args) {
        System.out.println("Press ENTER if you want to see a joke.");
        System.out.println("Type EXIT to exit.");
        Scanner sc_input = new Scanner(System.in);
        String input = sc_input.nextLine();
        
        tellJoke(input);
        sc_input.close();
    }

    public static void tellJoke(String input) {
        if (!input.equals("exit")) {
            System.out.println(ANSI_RED + "Why don't scientists trust atoms?");
            System.out.println(ANSI_GREEN + "Because they make up everything." + ANSI_RESET);
        }
    }
}